core/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -338,7 +344,18 @@ public ActionModule(boolean ingestEnabled, boolean transportClient, Settings set
         autoCreateIndex = transportClient ? null : new AutoCreateIndex(settings, clusterSettings, resolver);
         destructiveOperations = new DestructiveOperations(settings, clusterSettings);
         Set<String> headers = actionPlugins.stream().flatMap(p -> p.getRestHeaders().stream()).collect(Collectors.toSet());
-        restController = new RestController(settings, headers);
+        UnaryOperator<RestHandler> restWrapper = null;
+        for (ActionPlugin plugin : actionPlugins) {
+            UnaryOperator<RestHandler> newRestWrapper = plugin.getRestHandlerWrapper(threadPool.getThreadContext());
+            if (newRestWrapper != null) {
+                logger.debug("Using REST wrapper from plugin " + plugin.getClass().getName());
+                if (restWrapper != null) {
```

Comment:
I am not sure if we can do this in 5.x like this. There might be users relying on the filters... The other option is to just have multple wrappers and for iterate over them? passing the last wrapped one to the next wrapper?